### PR TITLE
Don't emit RETURN_SELF_NOT_MUST_USE lint if `Self` already is marked as `#[must_use]`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
New bug discovered with this lint. Hopefully, this is the last one.

---

changelog: none

### CODE REVIEW

File: clippy_lints/src/return_self_not_must_use.rs [hunk 1]
```diff
@@ -50,6 +51,7 @@ fn check_method(cx: &LateContext<'tcx>, decl: &'tcx FnDecl<'tcx>, fn_def: LocalD
         if decl.implicit_self.has_implicit_self();
         // We only show this warning for public exported methods.
         if cx.access_levels.is_exported(fn_def);
+        if !cx.tcx.hir().attrs(hir_id).iter().any(|attr| attr.has_name(sym::must_use));
```
Comment (xFrednet) [inline clippy_lints/src/return_self_not_must_use.rs:54]: Isn't this case covered by the `must_use_attr` check in line 57?

**Review by xFrednet — COMMENTED** [review]
One small question, it otherwise looks good to me :upside_down_face:

File: clippy_lints/src/return_self_not_must_use.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply clippy_lints/src/return_self_not_must_use.rs:54]: Indeed but apparently it's not working as it should. I'll clean up though. Good catch!

Comment (GuillaumeGomez) [reply clippy_lints/src/return_self_not_must_use.rs:55]: (Which was fixed in #8143 but I realized it would be better checked here instead.)

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by GuillaumeGomez** [comment]
And removed the warning as well.

**Review by xFrednet — APPROVED** [review]

**Comment by xFrednet** [comment]
LGTM, thank you for following up on these bugs :+1:. Let's hope it was the last one :upside_down_face: 

@bors r+

**Comment by GuillaumeGomez** [comment]
I hope as well! Thanks for the quick reviews!

**Comment by GuillaumeGomez** [comment]
@xFrednet Just to confirm: I still have this bug showing up when working on glib on the latest rust nightly:

```
$ rustc --version
rustc 1.59.0-nightly (e100ec5bc 2021-12-21)
$ cargo clippy --version
clippy 0.1.59 (e100ec5 2021-12-21)
```

Has this patch been "merged" into the nightly or not yet?

**Comment by xFrednet** [comment]
Hey @GuillaumeGomez, the fix has not been merged into nightly. The sync is usually done Thursdays, meaning today it should be synced. However, it might be delayed due to holidays and such. I can ping you on the sync PR or here if you like :upside_down_face:

**Comment by GuillaumeGomez** [comment]
Yes please! Thank you!

**Comment by flip1995** [comment]
The regular sync will be next week Thursday. I can do an out-of-cycle sync today, if you need this fix tomorrow.

**Comment by GuillaumeGomez** [comment]
No hurry, don't worry. Before beta/stable release would be nice though. ;)

**Comment by flip1995** [comment]
Yes, this will get in before beta branch 👍 (famous last words)

**Pull request merged**